### PR TITLE
Move NullLogger out of Internal

### PR DESCRIPTION
**Pull request by BrennanConroy**
#473 
Should NullScope also be moved?

### CODE REVIEW

File: src/Microsoft.Extensions.Logging.Abstractions/NullLogger.cs [hunk 1]
```diff
@@ -2,8 +2,9 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using Microsoft.Extensions.Logging.Abstractions.Internal;
 
-namespace Microsoft.Extensions.Logging.Abstractions.Internal
+namespace Microsoft.Extensions.Logging.Abstractions
 {
     public class NullLogger : ILogger
```
Comment (pranavkm) [inline src/Microsoft.Extensions.Logging.Abstractions/NullLogger.cs]: Docs since they're now public types.

**Comment by pranavkm** [comment]
:watch: for the docs.

**Comment by pranavkm** [comment]
> Should NullScope also be moved? 

Doesn't appear that it's publicly exposed. Plus it's more of a `NullDisposable` rather than a scope with the NullLogger, so leaving it internal seems ok to me..

**Comment by BrennanConroy** [comment]
Added some docs

File: src/Microsoft.Extensions.Logging.Abstractions/NullLogger.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    /// </summary>
+    public class NullLogger : ILogger
+    {
+        public static NullLogger Instance { get; } = new NullLogger();
+
+        private NullLogger()
+        {
+        }
+
+        /// <summary>
```
Comment (pranavkm) [inline src/Microsoft.Extensions.Logging.Abstractions/NullLogger.cs]: You could use `<inheritdoc />` for implementations. Should suffice in this case.

**Comment by pranavkm** [comment]
:shipit:

**Pull request merged**